Overhaul entire site and switch to pydata sphinx theme

djhoese opened the pull request:
This is a complete reorganization and restyling of the entire website. Main high-level changes:

1. Switch to pydata sphinx theme.
2. Create api docs as a separate subdirectory (`api/`) using `sphinx-apidocs`.
3. Include real quick start guides instead of just API docs.

TODO:

- [ ] Create a separate quick_start directory
- [ ] Create quick start doc for each interface (gloo, visuals (?), scene, plotting)
- [ ] Fix the huge number of warnings when generating these docs
- [ ] Figure out a good visualization for the home page (carousel still?)

Requires the related vispy repository changes (PR coming soon).

FIXES:

- [x] #15
- [x] #20
- [x] #21

djhoese committed:
Initial overhaul of documentation site

larsoner commented on api/vispy.app.tests.rst:
```diff
@@ -0,0 +1,70 @@
+   :undoc-members:
+   :show-inheritance:
+
+vispy.app.tests.test\_context module
```
This seems like a lot of boilerplate stuff. No way to have this done automatically via `autodoc` / `autosummary`?

larsoner replied:
(On the fly I mean, so no need to add it to the repo, just have it created during the sphinx build)

djhoese replied:
Ah at first I was like "this is `sphinx-apidocs`, does he think I wrote all this by hand?".

I've thought about that. Somehow even though I figured travis would update these automatically I didn't make the obvious connection that that would mean not storing them in the repository at all. That makes sense though.

It might be easiest to make `conf.py` generate this on the fly, thoughts? I figure it would be most compatible and most "automatic" for people running things manually.

djhoese replied:
Haha there's a package for doing it: https://pypi.org/project/sphinxcontrib-apidoc/#description

djhoese replied:
Done. Next commit in the main repository will include setup.py and travis updates.

djhoese committed:
Switch to sphinxcontrib.apidocs to dynamically generate the api docs

djhoese committed:
Fix link to modern GL book

djhoese committed:
Convert carousel from bootstrap 3 to bootstrap 4

djhoese commented:
@larsoner @kmuehlbauer This is where this is headed:

https://www.ssec.wisc.edu/~davidh/tmp/vispy/

I'm not sure I like how many things are in the top navigation bar, but am also not sure how to either combine them or what to remove. As more "guides" are created, I only see this getting worse.

larsoner commented:
Maybe fold "developer's guide" and "resources" into "community", rename "community" to "help" (or "documentation" or something), and make the order:

- Installation
- Getting started
- API
- Gallery
- Help
- News

Or something like this. I think it roughly follows the order of how people should use VisPy. Or for some stuff you can have it just on the landing page like [matplotlib](https://matplotlib.org/).

But in any case I agree there is too much stuff in the top bar.

kmuehlbauer commented:
@djhoese I've no other idea on the top bar. @larsoner :+1:

djhoese committed:
Add getting_started section

djhoese committed:
Add 'edit page' button

djhoese committed:
Restructure documentation again and create new navbar logo

djhoese commented:
Ok I just updated the site at https://www.ssec.wisc.edu/~davidh/tmp/vispy/. Looks something like this:

![image](https://user-images.githubusercontent.com/1828519/98413199-d2b3ec00-203e-11eb-99cb-e9e14cfb1a36.png)

Thoughts? In the future I'd like to throw more advanced/specific documentation (complex event, camera, or visual cases that don't fit in getting started) under the "Documentation" section, but then it gets a little weird to caption that as "Additional Help" as that would be more "Learning VisPy". I could put getting started under documentation but then it won't show up in the title navbar as a separate link. Overall I'm not completely sure on the separation or even the style of the captions for each section on the home page. All feedback welcome (typos, capitalization preferences, wording, etc).

My hope is to get a basic contributor guide started (make issues, make PRs, where to ask for help, docstring styling, how to check code style locally, etc) and maybe a few FAQs and then merge this and release it.

djhoese committed:
Add initial version of a contributor's guide

djhoese committed:
Add simple FAQ question

djhoese committed:
Add initial draft of a gloo getting started page

djhoese committed:
Finish first draft of gloo getting started

djhoese commented:
@kmuehlbauer @larsoner I've updated the copy of the site at https://www.ssec.wisc.edu/~davidh/tmp/vispy/ after making more changes to the gloo getting started. The other getting started's need to be completed but I think I'd like to merge this first so I can refer to it when helping users or requesting work from other people.

I also added a new FAQ about multithreading.

Feedback and concerns about merging early?

djhoese committed:
Cleanup gallery styling

djhoese committed:
More sphinx cleanup and add mention of citations

djhoese commented:
Kai gave his approval for an early merge of this PR with the promise that I'll make an issue for the remaining work that needs to be done.
